src/index.ts

```diff
@@ -373,25 +373,25 @@ export class UnleashClient extends TinyEmitter {
         return { ...this.context };
     }
 
-    public setContextField(field: string, value: string) {
+    public async setContextField(field: string, value: string): Promise<void> {
         if (isDefinedContextField(field)) {
             this.context = { ...this.context, [field]: value };
         } else {
             const properties = { ...this.context.properties, [field]: value };
             this.context = { ...this.context, properties };
         }
 
-        this.updateToggles();
+        await this.updateToggles();
```

Comment:
Maybe we should just return `this.updateToggles()` here? 
@kwasniew what do you think about this?

Reply:
In flutter we have Promise<void> equivalent: https://github.com/Unleash/unleash-flutter-sdk/blob/main/lib/unleash_proxy_client_flutter.dart#L350

I think we should incorporate this change as it is now